### PR TITLE
Document wildcards in build-external-contract

### PR DESCRIPTION
**Pull request by Utilitycoder**
This PR closes issues #719

### CODE REVIEW

File: website/docs/extensions/starknet/contract-target.md [hunk 1]
```diff
@@ -90,6 +90,35 @@ openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git", br
 build-external-contracts = ["openzeppelin::account::account::Account"]
 ```
 
+## Wildcard support in build-external-contracts
+
+The `[build-external-contracts]` property explained above allows asking Scarb to build contracts from package dependencies as part of proper package build, for example for use in tests. This release adds a possibility to request building many contracts from a module tree at once. For example, the following snippet:
```
Comment (maciektr) [inline website/docs/extensions/starknet/contract-target.md:95]:
```suggestion 
It is possible to request building many contracts from a module tree at once. For example, the following snippet:
```

1. I do not think there is a need to explain purpose of `build-external-contracts` here. 
1. Docs should not use expressions like "This release adds". It's not release notes.

File: website/docs/extensions/starknet/contract-target.md [hunk 2]
```diff
@@ -90,6 +90,35 @@ openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git", br
 build-external-contracts = ["openzeppelin::account::account::Account"]
 ```
 
+## Wildcard support in build-external-contracts
```
Comment (maciektr) [inline website/docs/extensions/starknet/contract-target.md:93]: Should be subparagraph of `Compiling external contracts. 
```suggestion
### Wildcard support
```

File: website/docs/extensions/starknet/contract-target.md [hunk 3]
```diff
@@ -90,6 +90,35 @@ openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git", br
...
+
+The `[build-external-contracts]` property explained above allows asking Scarb to build contracts from package dependencies as part of proper package build, for example for use in tests. This release adds a possibility to request building many contracts from a module tree at once. For example, the following snippet:
+
```
Comment (maciektr) [inline website/docs/extensions/starknet/contract-target.md:96]: Emphasize, that:
- Each external contract path can use upmost a single wildcard 
- A wildcard has to be the last character in contract path
- Express how matching is conducted - Scarb will take the contract path before wildcard and look for all contracts which paths start with it. 
Please reword appropriately.

File: website/docs/extensions/starknet/contract-target.md [hunk 4]
```diff
@@ -90,6 +90,35 @@ openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git", br
...
+    "dojo_erc::erc721::erc721::ERC721",
+    "dojo_erc::erc721::systems::erc721_approve",
+    "dojo_erc::erc721::systems::erc721_burn",
+    "dojo_erc::erc721::systems::erc721_mint",
+    "dojo_erc::erc721::systems::erc721_set_approval_for_all",
+    "dojo_erc::erc721::systems::erc721_transfer_from",
+]
+```
+
+can now be written as:
```
Comment (maciektr) [inline website/docs/extensions/starknet/contract-target.md:113]:
```suggestion
can be written as:
```
Docs do not refer to a single release changeset.

File: website/docs/extensions/starknet/contract-target.md [hunk 5]
```diff
@@ -90,6 +90,35 @@ openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git", br
...
+
+```toml-vue
+[[target.dojo]]
```
Comment (maciektr) [inline website/docs/extensions/starknet/contract-target.md:116]:
```suggestion
[[target.starknet-contract]]
```
Dojo target is not built into Scarb.

File: website/docs/extensions/starknet/contract-target.md (hunk 1 above)
Comment (Utilitycoder) [reply website/docs/extensions/starknet/contract-target.md]: I agree with taking that part out.

File: website/docs/extensions/starknet/contract-target.md (hunk 3 above)
Comment (Utilitycoder) [reply website/docs/extensions/starknet/contract-target.md:96]: Alright. I will work on rewording the paragraph.

File: website/docs/extensions/starknet/contract-target.md [hunk 6]
```diff
@@ -72,8 +72,10 @@ The off by default `casm-add-pythonic-hints` property enables Scarb to add it to
 
 While compiling the Scarb project, by default no artifacts are emitted for contracts defined in dependencies.
 To override this behaviour for certain contracts from other packages, you can use `build-external-contracts` property.
+
 It accepts a list of strings, each of which is a reference to a contract defined in a dependency.
 The package that implements this contracts need to be declared as a dependency of the project in `[dependencies]` section.
+
 The reference to a contract is a full cairo path to the contract module.
```
Comment (maciektr) [inline website/docs/extensions/starknet/contract-target.md:79]:
```suggestion
To override this behaviour for certain contracts from other packages, you can use `build-external-contracts` property.
It accepts a list of strings, each of which is a reference to a contract defined in a dependency.
The package that implements this contracts need to be declared as a dependency of the project in `[dependencies]` section.
The reference to a contract is a full cairo path to the contract module.
```
No need to add new lines here

File: website/docs/extensions/starknet/contract-target.md [hunk 7]
```diff
@@ -90,6 +92,37 @@ openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git", br
...
+can be written as:
+
+```toml-vue
+[[target.starknet-contract]]
+build-external-contracts = [
+    "dojo_erc::erc721::*",
+]
+```
+
+When using a wildcard in the build-external-contracts property, Scarb will match the contract path before the wildcard and look for all contracts whose paths start with that prefix. The wildcard can only be used as the last character in the contract path, and each external contract path can have at most one wildcard.
```
Comment (maciektr) [inline website/docs/extensions/starknet/contract-target.md:124]:
```suggestion
When using a wildcard in the `build-external-contracts` property, Scarb will match the contract path before the wildcard and look for all contracts whose paths start with that prefix. The wildcard can only be used as the last character in the contract path, and each external contract path can have at most one wildcard.
```

**Review by maciektr — APPROVED** [review]

**Pull request merged**